### PR TITLE
Handle 0x swap API errors

### PR DESCRIPTION
**Pull request by onyb**
In this PR we do a bunch of things:
* Retrieve error responses returned by the 0x API and log it in the console, for debugging purpose. We also add it to the redux store in order to handle specific errors that require user's attention.
* Perform some validation on the inputs before querying the 0x API. The idea is to prevent the possibility of running into errors while fetching quotes.
* Encode the validation/API error in the CTA button of Swap (inspired by Uniswap - see screenshots).

Here are the error types we support and the CTA text we use for them:
| CTA text | Error description | CTA button disabled |
-|-|-
|Insufficient balance| Token/ETH balance less than amount | Yes |
| Insufficient funds for gas | ETH balance insufficient for paying fees | Yes |
| Activate USDC (not implemented yet) | Insufficient ERC-20 token allowance | No |
| Insufficient liquidity | Not enough tokens in the liquidity pool to buy | Yes |
| Unknown error | Any other unhandled error returned by the 0x API | Yes |

Resolves https://github.com/brave/brave-browser/issues/18383.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Screenshots

|Screenshot 2021-10-01 at 9 47 58 PM|Screenshot 2021-10-01 at 9 48 18 PM|Screenshot 2021-10-01 at 9 49 03 PM|Screenshot 2021-10-01 at 10 45 25 PM|
-|-|-|-

### CODE REVIEW

File: components/brave_wallet_ui/components/buy-send-swap/swap/index.tsx [hunk 1]
```diff
@@ -75,6 +78,30 @@ function Swap (props: Props) {
     onFilterAssetList(toAsset)
   }
 
+  const submitText = React.useMemo(() => {
+    if (validationError === 'insufficientBalance') {
+      return 'Insufficient balance'
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/buy-send-swap/swap/index.tsx:83]: I think these should all be localized. 😎

**Review by Douglashdaniel — CHANGES_REQUESTED** [review]
Looks good! Just need to localize these error strings.

File: components/brave_wallet_ui/components/buy-send-swap/swap/index.tsx (hunk 1 above)
Comment (onyb) [reply components/brave_wallet_ui/components/buy-send-swap/swap/index.tsx]: Done!

File: components/brave_wallet_ui/components/buy-send-swap/swap/index.tsx [hunk 2]
```diff
@@ -75,6 +81,31 @@ function Swap (props: Props) {
...
+
+    if (validationError === 'insufficientLiquidity') {
+      return getLocale('braveWalletSwapInsufficientLiquidity')
+    }
+
+    if (validationError === 'unknownError') {
+      return getLocale('braveWalletSwapUnknownError')
+    }
+
+    return 'Swap'
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/buy-send-swap/swap/index.tsx:106]: `getLocale('braveWalletSwap')`

Comment (onyb) [reply components/brave_wallet_ui/components/buy-send-swap/swap/index.tsx]: Done!

**Review by Douglashdaniel — APPROVED** [review]
Looks good!

**Pull request merged**